ui/src/app/applications/components/applications-sync-panel/applications-sync-panel.tsx

```diff
@@ -2,23 +2,78 @@ import {ErrorNotification, FormField, NotificationType, SlidingPanel} from 'argo
 import * as React from 'react';
 import {Form, FormApi} from 'react-form';
 import {ARGO_WARNING_COLOR, ProgressPopup, Spinner} from '../../../shared/components';
-import {Consumer} from '../../../shared/context';
+import {Consumer, ContextApis} from '../../../shared/context';
 import * as models from '../../../shared/models';
 import {services} from '../../../shared/services';
 import {ApplicationRetryOptions} from '../application-retry-options/application-retry-options';
-import {ApplicationManualSyncFlags, ApplicationSyncOptions, FORCE_WARNING, REPLACE_WARNING, SyncFlags} from '../application-sync-options/application-sync-options';
+import {ApplicationManualSyncFlags, ApplicationSyncOptions, FORCE_WARNING, SyncFlags} from '../application-sync-options/application-sync-options';
 import {ApplicationSelector} from '../../../shared/components';
+import {confirmSyncingAppOfApps} from '../utils';
 
 interface Progress {
     percentage: number;
     title: string;
 }
 
+const InstanceLabel = 'app.kubernetes.io/instance';
+
 export const ApplicationsSyncPanel = ({show, apps, hide}: {show: boolean; apps: models.Application[]; hide: () => void}) => {
     const [form, setForm] = React.useState<FormApi>(null);
     const [progress, setProgress] = React.useState<Progress>(null);
     const getSelectedApps = (params: any) => apps.filter((_, i) => params['app/' + i]);
     const [isPending, setPending] = React.useState(false);
+    const [appList, setAppList] = React.useState(apps);
+    const syncHandler = (currentForm: FormApi, ctx: ContextApis, applications: models.Application[]) => {
+        const formValues = currentForm.getFormState().values;
+        const replaceChecked = formValues.syncOptions?.includes('Replace=true');
+        const selectedApps = [];
+        const selectedAppOfApps: models.Application[] = [];
+        let containAppOfApps = false;
+
+        for (const key in formValues) {
+            if (key.startsWith('app/') && formValues[key]) {
+                selectedApps.push(applications[parseInt(key.slice(key.lastIndexOf('/') + 1), 10)]);
+            }
+        }
+
+        selectedApps.forEach(app => {
+            if (app.isAppOfAppsPattern) {
+                containAppOfApps = true;
+                selectedAppOfApps.push(app);
+            }
+        });
+
+        if (replaceChecked && containAppOfApps) {
+            confirmSyncingAppOfApps(selectedAppOfApps, ctx, currentForm).then(confirmed => {
+                setPending(confirmed ? true : false);
+            });
+        } else {
+            currentForm.submitForm(null);
+        }
+    };
+
+    React.useEffect(() => {
+        const appOfAppsList: string[] = [];
+        apps.forEach(app => {
+            const appName = app.metadata.labels && app.metadata.labels[InstanceLabel];
```

Comment:
Is this the correct indicator of app of apps pattern?

Reply:
This label "determine which resources form the app", which could be "helm", "kustomize" or another app etc. It's not only for the app of apps pattern but if an app is created by another app, the resource app name is automatically generated to this label. 

This is the easiest way to tell if an app is App of Apps Pattern and I wanted to bring the solution up here for an open discussion and see if this is a reliable way to do it.

Reply:
@alexmt @rbreeze How do you think about this implementation? I tested with [/helm-guestbook](https://github.com/argoproj/argocd-example-apps/tree/master/helm-guestbook) and [/kustomize-guestbook](https://github.com/argoproj/argocd-example-apps/tree/master/kustomize-guestbook), none of these apps have the instance label in metadata.

Reply:
Update: checked out with Alex in the Argo CD Contributor's meeting. It's confirmed that `status.resources` list is a reliable field to check if an app is App of Apps Pattern. If there are more than 0 apps. in the `status.resources` field, it means this app is App of Apps Pattern. Will update this PR.